chain-impl-mockchain/src/header/header.rs

```diff
@@ -298,20 +298,25 @@ use chain_core::{
     property::{Deserialize, ReadError, Serialize},
 };
 
+impl SerializedSize for Header {
+    fn serialized_size(&self) -> usize {
+        0_u16.serialized_size() + self.as_slice().serialized_size()
+    }
+}
+
 impl Serialize for Header {
     fn serialize<W: std::io::Write>(&self, codec: &mut Codec<W>) -> Result<(), WriteError> {
-        codec.put_bytes(self.as_slice())
+        let bytes = self.as_slice();
+        codec.put_be_u16(bytes.len() as u16)?;
```

Comment:
I think `Header` is fixed size, no need to add length here. If deserializing without knowing the version (i.e. Praos, BFT, ...) is infeasible we could use a tag to define which `BlockVersion` this is